HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+
+        std::cout << "Original Rotation Matrix:\n" << rotationMatrix << std::endl;
+        std::cout << "New Rotation Matrix:\n" << rotationMatrixNew << std::endl;
+
+        // Calculate error in transformation (Rotation Matrix --> Roll-Pitch-Yaw --> Rotation Matrix)
+        std::cout << "\nVerifying Rotation Matrix" << std::endl;
+        compareRotation(rotationMatrix, rotationMatrixNew);
+
+        return 1;
+    }
+
+} // namespace TransformUtils
+
+int main()
+{
+    try
+    {
+        std::cout << "Opening .YAML file which contains transformation matrix (PoseState node)" << std::endl;
+        auto fileStorage = cv::FileStorage::FileStorage();
+        if(!fileStorage.open("..\\robotTransform.yaml", cv::FileStorage::Mode::READ))
+        {
+            std::cout << "Could not open file" << std::endl;
+        }
+        const auto poseStateNode = fileStorage["PoseState"];
+        std::cout << "Getting PoseState node" << std::endl;
+        if(poseStateNode.empty())
+        {
+            throw "PoseState node not found in file";
+        }
+        const auto poseState = poseStateNode.mat();
+        std::cout << poseState << std::endl;
+        fileStorage.release();
+
+        // Saving all conversions to .YAML file
+        if(!fileStorage.open("robotTransforms.yaml", cv::FileStorage::Mode::WRITE))
+        {
+            std::cout << "Could not open file for writing" << std::endl;
+        }
+        fileStorage.write("PoseState", poseState);
+
+        /*************************************************************************************************************
+         *      Direction Zivid to robot (Rotation Matrix --> any format)
+         *************************************************************************************************************/
+
+        // Transformation Matrix to Rotation Matrix and Translation Vector
+        std::cout << "\nConverting transformation matrix to rotation matrix and translation vector" << std::endl;
+        const Eigen::Matrix3d rotationMatrix = TransformUtils::cv2eigen(poseState(cv::Range(0, 3), cv::Range(0, 3)));
+        const Eigen::Vector3d translationVector = TransformUtils::cv2eigen(poseState(cv::Range(0, 3), cv::Range(3, 4)));
+        std::cout << "\nRotationMatrix:\n" << rotationMatrix << std::endl;
+        fileStorage.write("RotationMatrix", TransformUtils::eigen2cv(rotationMatrix));
+        std::cout << "\nTranslationVector:\n" << translationVector << std::endl;
+        fileStorage.write("TranslationVector", TransformUtils::eigen2cv(translationVector));
+
+        // Rotation Matrix in Transformation Matrix to Axis-angle
+        std::cout << "\nConverting rotation matrix to axis-angle" << std::endl;
+        Eigen::AngleAxisd axisAngle = Eigen::AngleAxisd(rotationMatrix);
+        Eigen::Vector4d axisAngleReordered = TransformUtils::reorderAxisAngle(axisAngle);
+        std::cout << "\nAxis-angle:\n" << axisAngleReordered << std::endl;
```

Comment:
I think we should print (ux uy uz theta) in brackets or under, the same way we write to file storage. It is not clear what is what. The same should be done for the quaternion and rotation vector. Perhaps even RPY, although it is mentioned in the function itself.